shared/ast_grep_utils.py

```
@@ -0,0 +1,80 @@
+def is_ast_grep_installed() -> bool:
+    """Check if ast-grep is available in PATH"""
+    return shutil.which('ast-grep') is not None
+
+def get_ast_grep_version() -> Optional[str]:
+    """Get installed AST-Grep version"""
+    if not is_ast_grep_installed():
+        return None
+
+    try:
+        result = subprocess.run(
+            ['ast-grep', '--version'],
```

Ruff (0.13.3)

Starting a process with a partial executable path

(S607)